File: id-card-toolkit-ios-sdk-v3.1.6/id-card-toolkit-ios-sdk-v3.1.6/samples/IDCardToolkitObjCApp/IDCardToolkitObjCApp/New Group/FamilyBookParse.m
//
//  FamilyBookParse.m
//  IDCardToolkitObjCApp
//
//  Created by Federal Authority For Identity and Citizenship on 19/12/17.
//  Copyright © 2017 Federal Authority For Identity and Citizenship. All rights reserved.
//

#import "FamilyBookParse.h"

@implementation FamilyBookParse

-(id)init {
    self=[super init];
    if (self) {
        
        self.familyBookKeyElements=[[NSMutableArray alloc]init];
        self.familyBookValueElements=[[NSMutableArray alloc]init];
    }
    return self;
}
-(void)getHeadOfFamilyDetails:(HeadOfFamily *)headofFamily {
    
    [self.familyBookKeyElements addObject:@"------HEAD OF FAMILY------"];
    [self.familyBookValueElements addObject:@" "];
    
    [self.familyBookKeyElements addObject:@"HolderIdNumber"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getHolderIdNumber]]];
    
    [self.familyBookKeyElements addObject:@"FamilyId"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getFamilyId]]];
    
    [self.familyBookKeyElements addObject:@"EmirateNameArabic"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getEmirateNameArabic]]];
    
    [self.familyBookKeyElements addObject:@"EmirateNameEnglish"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getEmirateNameEnglish]]];
    
    [self.familyBookKeyElements addObject:@"FirstNameArabic"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getFirstNameArabic]]];
    
    [self.familyBookKeyElements addObject:@"FirstNameEnglish"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getFirstNameEnglish]]];
    
    [self.familyBookKeyElements addObject:@"FatherNameArabic"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getFatherNameArabic]]];
    
    [self.familyBookKeyElements addObject:@"FatherNameEnglish"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getFatherNameEnglish]]];
    
    [self.familyBookKeyElements addObject:@"GrandFatherNameArabic"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getGrandFatherNameArabic]]];
    
    [self.familyBookKeyElements addObject:@"GrandFatherNameEnglish"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getGrandFatherNameEnglish]]];
    
    [self.familyBookKeyElements addObject:@"TribeArabic"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getTribeArabic]]];
    
    [self.familyBookKeyElements addObject:@"TribeEnglish"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getTribeEnglish]]];
    
    [self.familyBookKeyElements addObject:@"ClanArabic"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getClanArabic]]];
    
    [self.familyBookKeyElements addObject:@"ClanEnglish"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getClanEnglish]]];
    
    [self.familyBookKeyElements addObject:@"NationalityCode"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getNationalityCode]]];
    
    [self.familyBookKeyElements addObject:@"NationalityArabic"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getNationalityArabic]]];
    
    [self.familyBookKeyElements addObject:@"NationalityEnglish"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getNationalityEnglish]]];
    
    [self.familyBookKeyElements addObject:@"Gender"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getGender]]];
    
    [self.familyBookKeyElements addObject:@"DateOfBirth"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getDateOfBirth]]];
    
    [self.familyBookKeyElements addObject:@"PlaceOfBirthArabic"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getPlaceOfBirthArabic]]];
    
    [self.familyBookKeyElements addObject:@"PlaceOfBirthEnglish"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getPlaceOfBirthEnglish]]];
    
    [self.familyBookKeyElements addObject:@"MotherFullNameArabic"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getMotherFullNameArabic]]];
    
    [self.familyBookKeyElements addObject:@"MotherFullNameEnglish"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[headofFamily getMotherFullNameEnglish]]];
    
}
-(void)getWifeDetails:(Wife *)wife index:(int)index {
    
    [self.familyBookKeyElements addObject:[NSString stringWithFormat:@"------WIFE %i------",index+1]];
    [self.familyBookValueElements addObject:@" "];
    
    [self.familyBookKeyElements addObject:@"WifeIdNumber"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[wife getWifeIdn]]];
    
    [self.familyBookKeyElements addObject:@"FullNameArabic"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[wife getFullNameArabic]]];
    
    [self.familyBookKeyElements addObject:@"FullNameEnglish"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[wife getFullNameEnglish]]];
    
    [self.familyBookKeyElements addObject:@"NationalityCode"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[wife getNationalityCode]]];
    
    [self.familyBookKeyElements addObject:@"NationalityArabic"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[wife getNationalityArabic]]];
    
    [self.familyBookKeyElements addObject:@"NationalityEnglish"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[wife getNationalityEnglish]]];
}
-(void)getChildDetails:(Child *)child index:(int)index {
    
    [self.familyBookKeyElements addObject:[NSString stringWithFormat:@"------CHILD %i------",index+1]];
    [self.familyBookValueElements addObject:@" "];
    
    [self.familyBookKeyElements addObject:@"ChildIdNumber"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[child getChildIdn]]];
    
    [self.familyBookKeyElements addObject:@"FirstNameArabic"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[child getFirstNameArabic]]];
    
    [self.familyBookKeyElements addObject:@"FirstNameEnglish"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[child getFirstNameEnglish]]];
    
    [self.familyBookKeyElements addObject:@"Gender"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[child getGender]]];
    
    [self.familyBookKeyElements addObject:@"DateOfBirth"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[child getDateOfBirth]]];
    
    [self.familyBookKeyElements addObject:@"PlaceOfBirthArabic"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[child getPlaceOfBirthArabic]]];
    
    [self.familyBookKeyElements addObject:@"PlaceOfBirthEnglish"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[child getPlaceOfBirthEnglish]]];
    
    [self.familyBookKeyElements addObject:@"MotherIdNumber"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[child getMotherIdn]]];
    
    [self.familyBookKeyElements addObject:@"MotherFullNameArabic"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[child getMotherFullNameArabic]]];
    
    [self.familyBookKeyElements addObject:@"MotherFullNameEnglish"];
    [self.familyBookValueElements addObject:[NSString stringWithFormat:@"%@",[child getMotherFullNameEnglish]]];
}
-(NSMutableArray *)familyBookKey {
    return self.familyBookKeyElements;
}
-(NSMutableArray *)familyBookValue {
    return self.familyBookValueElements;
}

@end

File: id-card-toolkit-ios-sdk-v3.1.6/id-card-toolkit-ios-sdk-v3.1.6/samples/ICASwiftApp/ICASwiftApp/ConversionClass/FamilyBookParse.swift
//
//  FamilyBookParse.swift
//  IDCardToolkitSwiftApp
//
//  Created by Federal Authority For Identity and Citizenship on 23/12/17.
//  Copyright © 2017 Federal Authority For Identity and Citizenship. All rights reserved.
//

import UIKit

class FamilyBookParse: NSObject {
    
    var familyBookKeyElements = NSMutableArray()
    var familyBookValueElements = NSMutableArray()
    
    func getHeadOfFamilyDetails(_ headofFamily:HeadOfFamily) {
        
        var resultData:String
        
        self.familyBookKeyElements.add("---HeadofFamily---")
        self.familyBookValueElements.add(" ")
        
        resultData = headofFamily.getHolderIdNumber()
        self.familyBookKeyElements.add("HolderIdNumber")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getFamilyId()
        self.familyBookKeyElements.add("FamilyId")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getEmirateNameArabic()
        self.familyBookKeyElements.add("EmirateNameArabic")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getEmirateNameEnglish()
        self.familyBookKeyElements.add("EmirateNameEnglish")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getFirstNameArabic()
        self.familyBookKeyElements.add("FirstNameArabic")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getFirstNameEnglish()
        self.familyBookKeyElements.add("FirstNameEnglish")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getFatherNameArabic()
        self.familyBookKeyElements.add("FatherNameArabic")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getFatherNameEnglish()
        self.familyBookKeyElements.add("FatherNameEnglish")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getGrandFatherNameArabic()
        self.familyBookKeyElements.add("GrandFatherNameArabic")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getGrandFatherNameEnglish()
        self.familyBookKeyElements.add("GrandFatherNameEnglish")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getTribeArabic()
        self.familyBookKeyElements.add("TribeArabic")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getTribeEnglish()
        self.familyBookKeyElements.add("TribeEnglish")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getClanArabic()
        self.familyBookKeyElements.add("ClanArabic")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getClanEnglish()
        self.familyBookKeyElements.add("ClanEnglish")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getNationalityCode()
        self.familyBookKeyElements.add("NationalityCode")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getNationalityArabic()
        self.familyBookKeyElements.add("NationalityArabic")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getNationalityEnglish()
        self.familyBookKeyElements.add("NationalityEnglish")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getGender()
        self.familyBookKeyElements.add("Gender")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getDateOfBirth()
        self.familyBookKeyElements.add("DateOfBirth")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getPlaceOfBirthArabic()
        self.familyBookKeyElements.add("PlaceOfBirthArabic")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getPlaceOfBirthEnglish()
        self.familyBookKeyElements.add("PlaceOfBirthEnglish")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getMotherFullNameArabic()
        self.familyBookKeyElements.add("MotherFullNameArabic")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = headofFamily.getMotherFullNameEnglish()
        self.familyBookKeyElements.add("MotherFullNameEnglish")
        self.familyBookValueElements.add("\(resultData)")
        
    }
    func getWifeDetails(_ wife:Wife, index:Int) {
        
        self.familyBookKeyElements.add("------WIFE \(index+1)------ ")
        self.familyBookValueElements.add(" ")
        
        var resultData:String
        resultData = wife.getWifeIdn()
        self.familyBookKeyElements.add("WifeIdNumber")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = wife.getFullNameArabic()
        self.familyBookKeyElements.add("FullNameArabic")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = wife.getFullNameEnglish()
        self.familyBookKeyElements.add("FullNameEnglish")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = wife.getNationalityCode()
        self.familyBookKeyElements.add("NationalityCode")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = wife.getNationalityArabic()
        self.familyBookKeyElements.add("NationalityArabic")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = wife.getNationalityEnglish()
        self.familyBookKeyElements.add("NationalityEnglish")
        self.familyBookValueElements.add("\(resultData)")
        
    }
    func getChildDetails(_ child:Child, index:Int) {
        
        self.familyBookKeyElements.add("------CHILD \(index+1)------ ")
        self.familyBookValueElements.add(" ")
        
        var resultData:String
        resultData = child.getChildIdn()
        self.familyBookKeyElements.add("ChildIdNumber")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = child.getFirstNameArabic()
        self.familyBookKeyElements.add("FirstNameArabic")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = child.getFirstNameEnglish()
        self.familyBookKeyElements.add("FirstNameEnglish")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = child.getGender()
        self.familyBookKeyElements.add("Gender")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = child.getDateOfBirth()
        self.familyBookKeyElements.add("DateOfBirth")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = child.getPlaceOfBirthArabic()
        self.familyBookKeyElements.add("PlaceOfBirthArabic")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = child.getPlaceOfBirthEnglish()
        self.familyBookKeyElements.add("PlaceOfBirthEnglish")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = child.getMotherIdn()
        self.familyBookKeyElements.add("MotherIdNumber")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = child.getMotherFullNameArabic()
        self.familyBookKeyElements.add("MotherFullNameArabic")
        self.familyBookValueElements.add("\(resultData)")
        
        resultData = child.getMotherFullNameEnglish()
        self.familyBookKeyElements.add("MotherFullNameEnglish")
        self.familyBookValueElements.add("\(resultData)")
        
    }
    func familyBookKey() -> NSMutableArray {
        return self.familyBookKeyElements
    }
    func familyBookValue() -> NSMutableArray {
        return self.familyBookValueElements
    }
}

